internal/node/controller/init_test.go

```diff
@@ -0,0 +1,106 @@
+package controller
+
+import (
+	"encoding/json"
+	"fmt"
+	"testing"
+
+	"github.com/eagraf/habitat-new/core/state/node"
+	"github.com/eagraf/habitat-new/internal/node/config"
+	"github.com/eagraf/habitat-new/internal/node/hdb"
+	"github.com/spf13/viper"
+	"github.com/stretchr/testify/require"
+)
+
+func testTransitions(oldState *node.State, transitions []hdb.Transition) (*node.State, error) {
+	var oldJSONState *hdb.JSONState
+	schema := &node.NodeSchema{}
+	if oldState == nil {
+		emptyState, err := schema.EmptyState()
+		if err != nil {
+			return nil, err
+		}
+		ojs, err := hdb.StateToJSONState(emptyState)
+		if err != nil {
+			return nil, err
+		}
+		oldJSONState = ojs
+	} else {
+		ojs, err := hdb.StateToJSONState(oldState)
+		if err != nil {
+			return nil, err
+		}
+
+		oldJSONState = ojs
+	}
+	for _, t := range transitions {
+		if t.Type() == "" {
+			return nil, fmt.Errorf("transition type is empty")
+		}
+
+		err := t.Enrich(oldJSONState.Bytes())
+		if err != nil {
+			return nil, fmt.Errorf("transition enrichment failed: %s", err)
+		}
+
+		err = t.Validate(oldJSONState.Bytes())
+		if err != nil {
+			return nil, fmt.Errorf("transition validation failed: %s", err)
+		}
+
+		patch, err := t.Patch(oldJSONState.Bytes())
+		if err != nil {
+			return nil, err
+		}
+
+		newStateBytes, err := oldJSONState.ValidatePatch(patch)
+		if err != nil {
+			return nil, err
+		}
+
+		newState, err := hdb.NewJSONState(schema, newStateBytes)
+		if err != nil {
+			return nil, err
+		}
+
+		oldJSONState = newState
+	}
+
+	var state node.State
+	stateBytes := oldJSONState.Bytes()
+
+	err := json.Unmarshal(stateBytes, &state)
+	if err != nil {
+		return nil, err
+	}
+
+	return &state, nil
+}
+
+func TestInitTransitions(t *testing.T) {
+	config, err := config.NewNodeConfig()
+	require.Nil(t, err)
+
+	transitions, err := initTranstitions(config)
+	require.Nil(t, err)
+
+	require.Equal(t, 4, len(transitions))
```

Comment:
magic number? can we have a test method that counts transitions from a config?

Reply:
Same response here, I don't think we should get bogged down creating a constant for every number value in tests